pyanaconda/modules/payloads/payload/rpm_ostree/installation.py

```diff
@@ -633,6 +676,151 @@ def _set_kargs(self):
                  self._sysroot + "/boot"])
 
 
+class DeployBootcTask(Task):
+    """Task to deploy Bootc based image."""
+
+    def __init__(self, data, physroot, sysroot):
+        """Create a new task.
+
+        :param data: a bootc configuration
+        :param str physroot: a path to the physical root
+        :param str sysroot: a path to the system root
+        """
+        super().__init__()
+        self._data = data
+        self._physroot = physroot
+        self._sysroot = sysroot
+
+    @property
+    def name(self):
+        return "Deploy bootc"
+
+    def run(self):
+        bootloader = STORAGE.get_proxy(BOOTLOADER)
+        # Bootc will handle bootloader config so disable it
+        bootloader.BootloaderMode = 0
+        log.debug("Disabled bootloader configuration due to bootc mode")
+
+        stateroot = _get_stateroot(self._data)
+        ref = _get_ref(self._data)
+
+        log.debug("Run the bootc based installation")
+
+        self.report_progress(_("Bootc deployment starting: {}").format(ref))
+
+        # Some workarounds are needed due to current bootc development status
+
+        # The main one is SELinux to be presented in the system.
+        # https://github.com/bootc-dev/bootc/issues/1438
+        # It may be disabled but need to be presented.
+        # Both pairs of kernel parameters will work:
+        # * selinux=0 enforcing=0
+        # * selinux=1 enforcing=0
+        # Right now we use the /etc/selinux/config file to do the equivalent of
+        # selinux=1 enforcing=0.
+        # We set this in lorax-build using the whitelist_selinux.patch
+
+        # Bootc expects `prepare-root.conf` file to be presented in the system
+        # https://github.com/bootc-dev/bootc/discussions/1400
+        # https://github.com/bootc-dev/bootc/issues/1410
+        # echo -e "[ostree] \nsysroot=/sysroot" > /etc/ostree/prepare-root.conf
+        log.debug("Bootc workaround: add missing configuration file")
+        if not os.path.exists("/etc/ostree/prepare-root.conf"):
+            with open("/etc/ostree/prepare-root.conf", "w") as f:
+                f.write("[ostree]\n")
+                f.write("sysroot=/sysroot\n")
+        else:
+            log.debug("/etc/ostree/prepare-root.conf already presented and will not be modified")
+
+        # After automatic partitioning sysroot and sysimage are mounted,
+        # but we need a clear directory strucutre expected by the bootc
+        log.debug("Bootc workaround: remove unwanted mounts")
+        # umount -l /mnt/sysimage/
+        safe_exec_program("umount", ["-l", self._physroot])
+
+        # Bootc does not need any directories created automatically
+        # during the partitioning by blivet
+        log.debug("Bootc workaround: remove unwanted directories")
+        # rm -rf /mnt/sysroot/*
+        safe_exec_program("rm", ["-rf", self._sysroot + "/root"])
```

Comment:
Why is this different from the rest?

Reply:
Yeah, lets use the proper functions from `pyanaconda/core/path.py` or even add new one and use it here. :)